contrib/seccomp-notify-plugin-example-rust/src/mknod.rs

```diff
@@ -0,0 +1,183 @@
+use std::ffi::CStr;
+use std::ffi::CString;
+use std::os::unix::io::FromRawFd;
+use std::os::unix::io::AsRawFd;
+use std::os::unix::fs::FileExt;
+use std::io::Error;
+
+type Errno = i32;
+
+struct Device {
+    from: &'static str,
+    major: u32,
+    minor: u32,
+}
+
+const ALLOWED_DEVICES: [Device; 5] = [
+    Device{from: "/dev/null", major: 1, minor: 3},
+    Device{from: "/dev/zero", major: 1, minor: 5},
+    Device{from: "/dev/full", major: 1, minor: 7},
+    Device{from: "/dev/random", major: 1, minor: 8},
+    Device{from: "/dev/urandom", major: 1, minor: 9},
+];
+
+pub fn handle_mknodat_request(req: &mut nc::seccomp_notif_t) -> Result<bool, Errno> {
+    match handle_mknod_internal(0, 1, 2, 3, req) {
+        Ok(_) => {
+            Ok(false)
+        }
+        Err(errno) => {
+            if errno == libc::EPERM {
+                Ok(true)
+            } else {
+                Err(errno)
+            }
+        }
+    }
+}
+
+pub fn handle_mknod_request(req: &mut nc::seccomp_notif_t) -> Result<bool, Errno> {
+    match handle_mknod_internal(-1, 0, 1, 2, req) {
+        Ok(_) => {
+            Ok(false)
+        }
+        Err(errno) => {
+            if errno == libc::EPERM {
+                Ok(true)
+            } else {
+                Err(errno)
+            }
+        }
+    }
+}
+
+fn open_ro(path: String) -> Result<std::fs::File, Errno> {
```

Comment:
At this point, why aren't we just using [the standard open() API](https://doc.rust-lang.org/nightly/std/fs/struct.File.html#method.open)?

You'd just need to do something like `.map_err(|e| e.raw_os_err().unwrap_or(libc::EINVAL))` or so.

Reply:
thanks for the hint.  I've also changed the rest of the code to use `unwrap_or` instead of `unwrap`